src/components/HelpScoutBeacon/BeaconHeadScripts.js

```diff
@@ -1,67 +1,72 @@
-import React, { useEffect } from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
-import { Helmet } from 'react-helmet'
-import { noop, GU } from '../../utils'
+import { GU } from '../../utils'
+import { createGlobalStyle } from 'styled-components'
 
 const BEACON_EMBED =
   '!function(e,t,n){function a(){var e=t.getElementsByTagName("script")[0],n=t.createElement("script");n.type="text/javascript",n.async=!0,n.src="https://beacon-v2.helpscout.net",e.parentNode.insertBefore(n,e)}if(e.Beacon=n=function(t,n,a){e.Beacon.readyQueue.push({method:t,options:n,data:a})},n.readyQueue=[],"complete"===t.readyState)return a();e.attachEvent?e.attachEvent("onload",a):e.addEventListener("load",a,!1)}(window,document,window.Beacon||function(){});'
-const HELPSCOUT_ID = "'163e0284-762b-4e2d-b3b3-70a73a7e6c9f'"
-const BEACON_INIT = "window.Beacon('init'," + HELPSCOUT_ID + ')'
+const HELPSCOUT_ID = '163e0284-762b-4e2d-b3b3-70a73a7e6c9f'
 
-const BeaconHeadScripts = React.memo(({ optedIn, onReady }) => {
+function useHelpScoutBeacon(optedIn) {
+  const [beaconInit, setBeaconInit] = useState(false)
+
+  // Load the script if it doesn’t exist yet and optedIn
+  // is true, then set the value of Beacon.
   useEffect(() => {
-    let timeout = null
-    if (optedIn) {
-      ;(function isBeaconReady() {
-        if (window.Beacon) {
-          onReady()
-          return
-        }
-        timeout = setTimeout(isBeaconReady, 100)
-      })()
+    let script
 
-      return () => clearTimeout(timeout)
+    if (optedIn && !beaconInit) {
+      if (!window.Beacon) {
+        script = document.createElement('script')
+        script.innerHTML = BEACON_EMBED
+        document.body.appendChild(script)
+      }
+      window.Beacon('init', HELPSCOUT_ID)
```

Comment:
Super how this works!
I think this 2 actions (`init` and `setBeaconInit`) should be triggered in the `onload` method from script no?
Maybe attach the script to the head instead to the body?

Reply:
On another note, by removing the `beaconReady` prop the loader is no longer shown while the script is loading, you can test this by throttling the `network` speed in chrome, let's figure a way to show the `LoadingRing` to avoid the modal from disappearing and reappearing.

Reply:
> I think this 2 actions (init and setBeaconInit) should be triggered in the onload method from script no?

`Beacon.init()` gets queued and it’s probably a good idea to keep it first, as the script will probably refuse to execute the other methods if they are called before.

`beaconInit` is used to know that the script is being initialized, so that we don’t inject the script twice.